src/api/actions.rs

```diff
@@ -1,8 +1,53 @@
 //! GitHub Actions
 use snafu::ResultExt;
 
+use crate::models::{ArtifactId, RepositoryId, RunId, workflows::WorkflowDispatch};
 use crate::{params, Octocrab};
-use crate::models::{ArtifactId, RepositoryId, RunId};
+
+pub struct WorkflowDispatchBuilder<'octo> {
+    crab: &'octo Octocrab,
+    owner: String,
+    repo: String,
+    workflow_id: String,
+    data: WorkflowDispatch,
+}
+
+impl<'octo> WorkflowDispatchBuilder<'octo> {
+    pub(crate) fn new(crab: &'octo Octocrab, owner: String, repo: String, workflow_id: String) -> Self {
+        Self { crab, owner, repo, workflow_id, data: Default::default() }
+    }
+
+    /// Required
+    pub fn ref_field(mut self, ref_attr: impl Into<String>) -> Self {
+        self.data.ref_field = ref_attr.into();
+        self
+    }
```

Comment:
I'd remove `ref` as a separate method, as if it's a separate method, there's a chance you might miss it and only get that error at runtime, where as if we make it a required argument, this won't happen.

```suggestion
    pub(crate) fn new(crab: &'octo Octocrab, owner: String, repo: String, workflow_id: String, r#ref: String) -> Self {
        let mut this = Self { crab, owner, repo, workflow_id, data: Default::default() };
        this.data.r#ref = r#ref;
        this
    }
```